src/Native/Unix/System.IO.Ports.Native/pal_serial.c

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_types.h"
+#include <fcntl.h>
+#include <errno.h>
+#include <pal_serial.h>
+
+
+/* Open device file in non-blocking mode and without controlling terminal */
+int SerialPortOpen(const char * name)
```

Comment:
As noted in another comment, we should represent file descriptors at the interop layer as intptr_t.  As it's currently written, on a 64-bit platform there's potentially going to be corruption here, as this is going to return a 32-bit value and the managed code is expected a 64-bit value.